x-pack/plugins/watcher/server/models/action_status/action_status.js

```diff
@@ -30,7 +32,10 @@ export class ActionStatus {
     const actionStatusJson = this.actionStatusJson;
     const ackState = actionStatusJson.ack.state;
 
-    if (this.lastExecutionSuccessful === false) {
+    if (
+      this.lastExecutionSuccessful === false &&
+      this.lastCheckedRawFormat === this.lastExecutionRawFormat
+    ) {
```

Comment:
From this it looks like `lastExecutionSuccessful` can be `false` but we have an edge case where `lastCheckedRawFormat` come after that? I.o.w. the last execution could be successful even though the lastExecutionSuccessful field is false?

Reply:
So the watch only "executes" when the condition is met, but the watch condition is checked on a given interval. So the last time it was executed could be false, but the next time it is checked the condition may not be met so it doesn't execute. Does that make sense? I also tried to explain this in the PR description, but I'm probably not doing the best job so let me know if you need me to clarify more 😅 .